internal/pkg/template/templates/environment/cf.yml

```diff
@@ -38,7 +38,12 @@ Parameters:
     AllowedValues: [true, false]
   ServiceDiscoveryEndpoint:
     Type: String
+  ImportedVPCID:
+    Type: String
+  ImportedPrivateSubnets:
+    Type: String
```

Comment:
Hmmm, what do you think of a different alternative where we do option2. but generate it for the user.

i.e. when a user creates an Addon, we can create a `mappings.yml` file that contains a Mappings like:
```yaml
Mappings:
  VpcId:
      test: vpc-12345
      prod: vpc-78912
```

My intuition is that we want stack parameters to be generic Parameters rather than specific like `ImportedVPCID`.
For example, we can have just a `VpcId` parameter where:
- If empty, we create the VPC.
- Otherwise, we use the value.


So I'd be OK with either generating the `mappings` file or using flexible Parameters.

Reply:
```
Mappings:
  VpcId:
      test: vpc-12345
      prod: vpc-78912
```
People would need to manage an imported VPC IDs in both the env manifests and ⬆️.

> flexible Parameters

Just making sure I'm on the same page with ya - 

In addition to renaming `ImportedVPCID` to `VpcId`, we want to change the creation of VPC-related resources from using `go-template` rendering to using CFN `Conditions` - is this what this proposal implies? I'm down to do so

Reply:
>  is this what this proposal implies?

Yeah that's what I meant, but to be honest I am more aligned with @iamhopaul123 's recommendation of not doing anything until a customer asks for it because there is a mitigation path

Reply:
Yeah after a second thought i feel like it's probably fine to have users who import VPC/subnets to hardcode the ID. It's not a very popular use case to import VPC/subnets and use them in addon stack at the same time, but adding conditions and params is a one-way door 🤔